### DISCUSSION TITLE
Probabilistic damage state calculation for earthquakes

### DISCUSSION THREAD

**Discussion by hkayabilisim** [TomorrowsCities/tomorrowscities#118] [Ideas]
In the current version of the WebApp (as of v0.6), the damage state of a building is calculated based on largest vertical distance between fragility curves for a given ground shaking intensity. In this discussion, we are investigating a new probabilistic implementation based on a [Psuedo-code](https://github.com/user-attachments/files/16438725/Psuedo-code.for.the.DS.docx)

**Comment by hkayabilisim** [comment]
The Psuedo-code is very clear and easy to implement. There are few things need to be decided:

* Are we going to provide this new implementation as a second alternative which the users can select? Or do we completely wipe out the old implementation?
* In the Psuedo-code, it is mentioned that the implementation should work even if there are less then four fragility curves. It would be good to have an example scenario for this situation.
* For a given building, are we going to apply the code for a single time or repeat the process for let's say 100 times and report the most frequent damage state?

**Reply by karimaljawhari** [reply]
Thank you Huseyin; here are my thoughts:

- The new modification is more statistically sound as it produces a potential realization of damage states rather than being constrained to the most probable damage state (the one with the largest vertical distance). We can keep both options for now, and decide later if we would keep both or just use the new one.

- The pseudo-code has now been modified to account for the flexibility in the number of damage states (see attached)
[Psuedo-code for the DS_flexible_DS.docx](https://github.com/user-attachments/files/16440555/Psuedo-code.for.the.DS_flexible_DS.docx)

- The random selection of damage state will be applied only once because if we do it a large number of times (say 1000 random selections) to reach convergence, then we need to select the damage state that occurred the most, which will be the one with highest probability (i.e., largest vertical distance)

**Comment by hkayabilisim** [comment]
Thank you for the updated pseudo-code. I wonder if we can skip converting the probabilities(i.e. 100*(1-p)), and work directly on probability of exceedence (POE). See, the mapping function below. If it looks good, we can start implementing.

![IMG_0131](https://github.com/user-attachments/assets/0a7e9d12-ec91-450b-b73e-69b193b7288f)

**Reply by karimaljawhari** [reply]
Hi Huseyin,
Thanks for the great comment! Yes it is possible, however, there are two things to keep in mind:

1. The random number should be between 0 and 1 rather than 0 to 100, as you wrote in the figure.
2. For the case of four damage states, the following happens:

     RN <   PoE(DS4), then we have DS4
     RN >= PoE(DS4) and RN < PoE(DS3), then we have DS3
     RN >= PoE(DS3) and RN < PoE(DS2), then we have DS2
     RN >= PoE(DS2) and RN < PoE(DS1), then we have DS1
     RN >   PoE(DS1), then we have DS0

This is exactly what you described visually in the figures you attached.

Thanks a lot!

**Comment by karimaljawhari** [comment]
An update for the computational algorithm of the WebApp:

- The code provided here brings us a single realization of the damage for each building in the region we are looking at
- The code will be soon extended so that we can conduct multiple runs (say 500 for example), in each run, there will be a scenario of potential impacts. 
- From the 500 runs, we can shed light on some probabilistic indicators (e.g., mean, median, 16th and 84th percentiles, etc.).
